src/api/product/product.types.ts

```diff
@@ -0,0 +1,10 @@
+export type ProductItemProps = {
```

Comment:
Nie do końca o to mi chodziło :D
W API są typy ogólne, do tzw encji (słownictwo z backendu, z baz danych), oraz otypowanie requestów i responsów.
Jeśli chodzi o te encje, to są takie typy, które definiują jakieś "twory"/"obiekty", takie jak np `User`, `Product` itd.
Więc w tym przypadku ma się tutaj znaleźć typ taki:
```
export type Product = {
    id: string;
    name: string;
    condition: string;
    price: number;
    category: string;
    photo: string;
};
```
A przy `ProductItem` ma dalej być:
```
export type ProductItemProps = {
  productDetails: Product;
}
```